enhancements/builds/volume-mounted-resources.md

```diff
@@ -0,0 +1,297 @@
+  // Defaults to false.
+  // +optional
+  ReadOnly bool
+  // Path within the build environment at which the volume should be mounted.  Must
+  // not contain ':'.
+  MountPath string
+}
+```
+
+Example usage (use of the existing secret/configmap injection api is included for comparison, it is not changing)
+
+```yaml
+apiVersion: v1
+items:
+- apiVersion: build.openshift.io/v1
+  kind: BuildConfig
+  metadata:
+    name: mybuild
+    namespace: p1
+  spec:
+    strategy:
+      sourceStrategy:
+        from:
+          kind: ImageStreamTag
+          name: nodejs:10-SCL
+          namespace: openshift
+        volumes:
+        - name: secret
+          secret:
+            secretName: somesecret
+        - name: config
+          configMap:
+            name: someconfigmap
+            items:
+            - key: somekey
+              path: volume/path/value.txt
+        - name: content-access
+          csi:
+            driver: csi-driver-projected-resource.openshift.io
+            volumeAttributes:
+              share: etc-pki-entitlement
+        volumeMounts:
+        - name: config
+          mountPath: /tmp/config
+        - name: secret
+          mountPath: /tmp/secret
+        - name: content-access
+          mountPath: /etc/pki/entitlement
+      type: Source
+    source:
+      secrets:
+      - secret: 
+          name: myOtherSecret
+        destinationDir: /tmp/othersecret
+      configMaps:
+      - configMap:
+          name: myOtherConfigMap
+        destinationDir: /tmp/otherconfig
+```
+
+### Risks and Mitigations
+
+Since the build pod is privileged, we need to ensure that users cannot abuse this api to trick the build controller into creating build pods that can exploit those privileges.
+This means ensuring that any volume mount specifications the user provides, which are translated into volumemounts in the build pod, cannot be used to alter the build logic.
+To this end, we should explicitly control where the volumes are mounted within the build pod.(We can mount them anywhere the user specifies within the buildah container).
+
+We also need to ensure that the user can't use this api to inject/mount content that they could not normally mount into a pod they created themselves.
+For this reason we must explicitly disallow `HostPath` volume types, for example.
+We will mitigate this by only supporting a subset of VolumeSources that can be mounted into builds.
+As additional types are requested, we will need to determine it is safe to add them.
+Gating will also ensure that we address any volume lifecycle concerns (for example - binding of PersistentVolumeClaims).
+
+## Design Details
+
+### Test Plan
+
+This feature will need new e2e tests that leverage the new api option.
+We have existing tests for configmap+secret injection would should be able to be copied+adapted to testing this feature relatively easily.
+Tests for the CSI volume may have to be done independently of the core OpenShift test suite at outset.
```

Comment:
Yep until the move into the CVO via the storage operators happens, once openshift/api and openshift/client-go are updated with the new fields, we can vendor them into https://github.com/openshift/csi-driver-projected-resource and build upon the existing e2e's there to create BC's that use the new stuff.